alodataset/transforms.py

```diff
@@ -858,3 +858,43 @@ class RandomDownScaleCrop(Compose):
     def __init__(self, size, preserve_ratio=False, *args, **kwargs):
         transforms = [RandomDownScale(size, preserve_ratio), RandomCrop(size)]
         super().__init__(transforms, *args, **kwargs)
+
+
+class DynamicCropTransform(AloTransform):
+    """ Crop image to target crop size at chosen position.
+    """
+    def __init__(self, crop_size, *args, **kwargs):
+        assert all([isinstance(s, int) for s in crop_size])
+        self.crop_size = crop_size
+
+        super().__init__(*args, **kwargs)
+
+    def sample_params(self):
+        return (self.crop_size,)
+
+    def set_params(self, size):
+        self.crop_size = size
+
+    def apply(self, frame: Frame, center: Union[Tuple[int, int], Tuple[float, float]] = (0.5, 0.5)):
+        """
+        center: Coordinate of cropped image center. This coordinate is tuple of int or tuple of float.
+                Default: (0.5, 0.5)
+        """
+        if isinstance(center[0], float):
+            center_x = int(frame.W * center[0])
+            center_y = int(frame.H * center[1])
+        else:
+            center_x = center[0]
+            center_y = center[1]
+
+        crop_h, crop_w = self.crop_size[0], self.crop_size[1]
+        left = center_x - crop_w // 2
+        top = center_y - crop_h // 2
+        right = center_x + crop_w // 2
+        bot = center_y + crop_h // 2
+
+        if left < 0 or top < 0 or right > (frame.W - 1) or bot > (frame.H - 1):
```

Comment:
I don't know if that's gonna be useful or not but ValueError raises when W or H is the same as one of the corresponding crop sizes (i.e if we want to keep the same frame's H or W and crop the other dim)
Exemple :

```python
center = [0.5, 0.5]
frame.W = 50
frame.H = 20
crop_size = [20, 20]
```

```python
ValueError: Crop coordinates out of image border. Image size: (50, 20), Crop coordinate (top, left, bot, right): (0, 15, 20, 35)
```